graph/src/runtime/asc_ptr.rs

```diff
@@ -150,12 +150,13 @@ impl<C: AscType> AscPtr<C> {
     /// This function returns the `rt_size`.
     /// Only used for version >= 0.0.5.
     pub fn read_len<H: AscHeap + ?Sized>(&self, heap: &H) -> Result<u32, DeterministicHostError> {
+        // We're trying to read the pointer below, we should assert it's
+        // not null before using it.
+        self.assert_is_not_null()?;
+
         let size_of_rt_size = 4;
         let start_of_rt_size = self.0.checked_sub(size_of_rt_size).ok_or_else(|| {
-            DeterministicHostError(anyhow::anyhow!(
-                "Subtract overflow on pointer: {}", // Usually when pointer is zero because of null in AssemblyScript
-                self.0
-            ))
+            DeterministicHostError(anyhow::anyhow!("Subtract overflow on pointer: {}", self.0))
```

Comment:
Could we say something more helpful here (like what that means in terms of pointers)? Is this always an internal error? And it seems that `size_of_rt_size` could be `const`.

Reply:
You're correct, the size of `rt_size` can be changed to a `const`, I'll change that.

This `Subtract overflow` shouldn't happen, it's related to the way every type is layed out in memory.

All class instances in AssemblyScript have a header of 20 bytes, of which the last 4 hold the size of the following data.

AS header memory layout docs: https://www.assemblyscript.org/memory.html#common-header-layout.

Our `AscPtr` structure holds a number which is the address in WebAssembly memory of where that class instance starts their data (right after the header).

So this part of the code tries to do a pointer arithmetic to get the size of the structure, so it can read it fully in a correct fashion.

Maybe what I can write instead of this message is that it failed to find the size of the following structure.

Reply:
I added the constant but I'll let these (there are more) "Subtract overflow" to fix later. I want to discuss more with @leoyvens on this.